zookeeper-server/src/main/java/org/apache/zookeeper/common/X509Util.java

```diff
@@ -548,17 +548,19 @@ public static X509TrustManager createTrustManager(
         try {
             KeyStore ts = loadTrustStore(trustStoreLocation, trustStorePassword, trustStoreTypeProp);
             PKIXBuilderParameters pbParams = new PKIXBuilderParameters(ts, new X509CertSelector());
-            if (crlEnabled || ocspEnabled) {
-                pbParams.setRevocationEnabled(true);
-                System.setProperty("com.sun.net.ssl.checkRevocation", "true");
-                System.setProperty("com.sun.security.enableCRLDP", "true");
-                if (ocspEnabled) {
-                    Security.setProperty("ocsp.enable", "true");
-                }
-            } else {
-                pbParams.setRevocationEnabled(false);
+            // Leave CRL/OCSP JVM global properties alone both are set to "system" (represented as null)
+            if (!crlEnabled.isSystem() || !ocspEnabled.isSystem()) {
+	            if (crlEnabled.isTrue() || ocspEnabled.isTrue()) {
+	                pbParams.setRevocationEnabled(true);
+	                System.setProperty("com.sun.net.ssl.checkRevocation", "true");
+	                System.setProperty("com.sun.security.enableCRLDP", "true");
+	                if (ocspEnabled.isTrue()) {
+	                    Security.setProperty("ocsp.enable", "true");
+	                }
+	            } else {
+	                pbParams.setRevocationEnabled(false);
+	            }
```

Comment:
One question here:
In the case when both parameters are set to 'system', we won't do anything: not altering system properties and not setting revocation in `pbParams`. Will that work correctly, I mean will `pbParams` revocation flag follow the system settings?

Reply:
Doc:
> When a PKIXParameters object is created, this flag is set to true. This setting reflects the most common strategy for checking revocation, since each service provider must support revocation checking to be PKIX compliant. Sophisticated applications should set this flag to false when it is not practical to use a PKIX service provider's default revocation checking mechanism or when an alternative revocation checking mechanism is to be substituted (by also calling the addCertPathChecker or setCertPathCheckers methods).

Reply:
I found the same docs, @anmolnar .

The code agrees that this defaults to the value of "com.sun.net.ssl.checkRevocation".

https://github.com/openjdk/jdk/blob/7c13a2cd9aa5ec9da00084de2388abc189e2f4ef/src/java.base/share/classes/sun/security/validator/PKIXValidator.java#L174

I think that's fine, the goal here is to use the JDK default settings.
This one CAN be overriden separately with the new property if we use a custom truststore.

(Also note that I have slightly changed the logic in the current #2277 PR).